python/taichi/lang/ops.py

```diff
@@ -7,7 +7,26 @@
 from taichi.lang import expr, impl
 from taichi.lang.exception import TaichiSyntaxError
 from taichi.lang.field import Field
-from taichi.lang.util import cook_dtype, is_taichi_class, taichi_scope
+from taichi.lang.util import (cook_dtype, is_matrix_class, is_taichi_class,
+                              taichi_scope)
+
+
+def uniform_matrix_inputs(*args):
+    has_real_matrix = False
+    for arg in args:
+        if is_taichi_expr(arg) and arg.ptr.is_tensor():
+            has_real_matrix = True
+            break
+
+    results = []
+    for arg in args:
+        if has_real_matrix and is_matrix_class(arg):
```

Comment:
Can we simply use `isinstance(arg, Matrix)` instead of adding `is_matrix_class`?

Reply:
Interesting, the root reason is that Python doesn't support separate declaration and definition, therefore circular import can be a real headache. 

However, if it supports separate declaration and definition, we'll then suffer from symbol inconsistency instead :(